src/sentry/utils/session_store.py

```diff
@@ -13,6 +13,13 @@ class RedisSessionStore:
     the request session. Useful for storing data too large to be stored into
     the session cookie.
 
+    The attributes to be backed by Redis must be declared in a subclass using
+    the `redis_proerty` function. For example:
```

Comment:
Changing from dynamic attributes was motivated primarily by weird errors I kept hitting while testing, where the `redis_key` or `auth_key` or something would go to the `__getattr__` method (thereby causing infinite recursion) instead of to the property. Even if those bugs are avoidable, I would defend this change.

**Pro:**

- Avoids surprising differences between Redis-backed attributes from regular attributes.
- Exposes intent up front of what is to be stored in Redis.
- Allows IDEs to see and jump to (e.g., Cmd+Click) a declared property definition.
- Prevents risk of bugs from `__getattr__` and `__setattr__` being hit unintentionally.

**Con:**

- Necessitates subclasses to add properties for individual use cases.
- Existing references to attributes not covered this way will cause `AttributeError` (but tests should catch all of those).

We could avoid the subclasses by pulling the full universe of properties up into the base class, but this class should be a general tool and I don't think concerns about the attributes to be stored (e.g., "flow") belong up here.